packages/pyms/basic/p_partial_timeseries.py

```diff
@@ -0,0 +1,114 @@
+import numpy as np
+
+import sys,os
+parent_path=os.path.dirname(os.path.dirname(os.path.abspath(__file__)))
+sys.path.append(parent_path)
+
+from mlpy import writemda32,writemda64,readmda,DiskReadMda,DiskWriteMda,MdaHeader
+from common import TimeseriesChunkReader
+
+processor_name='pyms.partial_timeseries'
+processor_version='0.10'
+
+def partial_timeseries(*,timeseries,timeseries_out_all=[],t1_all=[],t2_all=[],
+                       timeseries_dtype='',timeseries_num_channels=4):
+    """
+    (Different from pyms.p_extract_timeseries, this function reads in a timeseries, a list of query data time points, 
+        and output a list of cut files.)
+    Extract a chunk of a timeseries dataset with events between A LIST OF t1's and t2's.
+
+    Parameters
+    ----------
+    timeseries : INPUT
+        Path of firing.mda timeseries (as it uses time ind in the second row to index how to cut), 
+        If non-binary, data should be in shape of MxN where M is number of channels and N number of timepoints, in either .mda or raw binary format. 
+        If raw binary, then you must supply dtype.
+    timeseries_out_all: OUTPUT
+        A list of paths of output timeseries in .mda format
+    t1_all: list of integer, start timepoints (zero-based indexing).
+    t2_all: list of integer, Integer end timepoints (zero-based indexing),
+    timeseries_dtype : string
+        Only supply this if timeseries is in raw binary format. Choices are int16, uint16, int32, float32, etc.
+    timeseries_num_channels : integer
+        Only supply this if timeseries is in raw binary format. Integer representing number of channels. Number of timepoints will be deduced
+    """
+
+    
+    header0=None
+    if (timeseries_dtype):
+        size_bytes=os.path.getsize(timeseries)
+        num_bytes_per_entry=get_num_bytes_per_entry_from_dt(timeseries_dtype)
+        num_entries=size_bytes/num_bytes_per_entry
+        if (num_entries % timeseries_num_channels != 0):
+            print ("File size (%ld) is not divisible by number of channels (%g) for dtype=%s" % (size_bytes,timeseries_num_channels,timeseries_dtype))
+            return False            
+        num_timepoints=num_entries/timeseries_num_channels
+        header0=MdaHeader(timeseries_dtype,[timeseries_num_channels,num_timepoints])
+    
+    X=DiskReadMda(timeseries,header0)
+    M,N = X.N1(),X.N2()
+
+    Xdata=X.readChunk(i1=0,i2=0,N1=M,N2=N)
+        
+    chunk_size_mb=100
+
+    for i in range(0,len(t1_all)):
+        t1=t1_all[i]
+        t2=t2_all[i]
+        timeseries_out=timeseries_out_all[i]
+
+        t1t2=np.nonzero(np.logical_and([Xdata[1,:]>=t1],[Xdata[1,:]<=t2]))
```

Comment:
Extract `Xdata[1,:]` as a separate variable — this might be splicing the array twice now.